.github/workflows/codeql.yml

```
@@ -0,0 +1,77 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main", "[0-9].[0-9]", "[0-9].x" ]
```

What happen with the branch 4.12? Don't match with this pattern.
```suggestion
    branches: [ "main", "[0-9].[0-9]+", "[0-9].x" ]
```

---

This pr is to update the base branch as it was currently in 4.12.0.